Upgrade Deadline Repo and DocDB; Fix Integration Tests

loachri opened the pull request:
We have broken Integration tests for RFDK. The primary issue was due expired rds-ca-2019 CA certs. The fix was to upgrade the tests to use newer versions of MongoDB and DocDB, which use the newer rds-ca-rsa2048-g1 CA certs, as well as fix any downsteam issue that came as a result of upgrading MongoDB and DocDB

Deadline Repository was upgraded to 10.3.2.1, and DocDB has been upgraded to 5.0.0.

I've also increased timeouts in the repo installer as well as in the jest settings, as a considerable amount of test flakeyness was a result of these timeouts.

Finally, I've added more verbose debugging to the DeadlineRepository installer, so that it's more clear what's going on if it fails to install

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

rondeau-aws commented on integ/components/deadline/deadline_01_repository/scripts/bastion/testing/DL-read-docdb-response.sh:
```diff
@@ -39,6 +39,5 @@ else
 # Mongo command to query for "deadline10db" database
-mongo --quiet --ssl --host="$DB_ADDRESS" --sslCAFile="$CERT_CA" --username="$DB_USERNAME" --password="$DB_PASS" --eval='printjson( db.adminCommand( { listDatabases: 1, nameOnly: true, filter: { "name": "deadline10db" } } ) )'
-
+mongo --quiet --ssl --host="$DB_ADDRESS" --sslCAFile="$CERT_CA" --username="$DB_USERNAME" --password="$DB_PASS" --eval='(function(){var output=db.adminCommand({ listDatabases: 1, nameOnly: true, filter: { "name": "deadline10db" } } );delete output.onTime;delete output.operationTime;printjson(output)})()'
```
This line is getting rather long and difficult to read.  Would it be possible to split it up?  I think our style guide for this project recommends a line length of 120 characters.

Also, should we document why we added these `delete` commands?

loachri replied:
I'm okay with just leaving this given that it's a shell script, and I don't want to jeopardize some sort of weirdness by splitting up the bash command into different lines.

I can add a comment here

rondeau-aws commented on packages/aws-rfdk/lib/deadline/scripts/bash/installDeadlineRepository.sh:
```diff
@@ -183,7 +183,7 @@ fi
 # substituted in its place. If "parameter" is set, then the value of "word" is used, which is the expansion of the populated array.
 # Since bash treats the expansion of an empty array as an unset variable, we can use this pattern expand the array only if it is populated.
-$INSTALL_AS_NON_ROOT_CMD $REPO_INSTALLER --mode unattended --setpermissions false --prefix "$PREFIX" --installmongodb false --backuprepo false ${REPO_ARGS[@]+"${REPO_ARGS[@]}"}
+$INSTALL_AS_NON_ROOT_CMD $REPO_INSTALLER --mode unattended --setpermissions false --prefix "$PREFIX" --installmongodb false --backuprepo false --debuglevel 4 ${REPO_ARGS[@]+"${REPO_ARGS[@]}"}
```
It's probably fine, but I just want to make sure: did you intend to include the `--debuglevel 4` here?

loachri replied:
Yes, the installbuilder will not give any indication of failure otherwise

rondeau-aws commented on packages/aws-rfdk/lib/deadline/lib/repository.ts:
```diff
@@ -677,7 +678,7 @@ export class Repository extends Construct implements IRepository {
       const parameterGroup = databaseAuditLogging ? new ClusterParameterGroup(this, 'ParameterGroup', {
         description: 'DocDB cluster parameter group with enabled audit logs',
-        family: 'docdb3.6',
+        family: 'docdb5.0',
```
We should probably call out the update to DocDB 5.0 in the release notes?

As far as I know, the release notes are automatically generated based on commit messages.  Unfortunately I don't know if that can handle multiple notes in a single commit, or if we'd need to split up the commits.

loachri replied:
I can see if I can split this up into multiple commits

loachri replied:
Split up the commit into multiple commits

rondeau-aws commented on integ/components/deadline/deadline_01_repository/lib/repository-testing-tier.ts:
```diff
@@ -115,7 +115,7 @@ export class RepositoryTestingTier extends TestingTier {
       'mkdir -p testScripts',
       'cd testScripts',
-      'wget https://s3.amazonaws.com/rds-downloads/rds-combined-ca-bundle.pem',
+      'wget https://truststore.pki.rds.amazonaws.com/global/global-bundle.pem',
```
We should probably call this out in the release notes / CR description too?

loachri replied:
I'll add this in

loachri replied:
Added mention of this to the commit notes. There's some mention of this in the CR description, since this is the code that uses the new CA cert rather than the expired 2019 one

rondeau-aws commented on packages/aws-rfdk/lib/deadline/lib/repository.ts:
```diff
@@ -686,7 +687,7 @@ export class Repository extends Construct implements IRepository {
       const dbCluster = new DatabaseCluster(this, 'DocumentDatabase', {
         masterUser: {username: 'DocDBUser'},
-        engineVersion: '3.6.0',
+        engineVersion: '5.0.0',
```
Looks like we're still using DocumentDB 3.6 in a few places such as `examples/deadline/All-In-AWS-Infrastructure-Basic/ts/lib/storage-tier.ts`.  Do we need to update those too?

loachri replied:
I'll update this

loachri replied:
Actually, I'll just update the example, since I don't want to update the integ structs without doing a retest, and I don't think we have bandwidth for a retest

loachri replied:
Updated this

loachri committed:
feat(deadline)!: upgrade deadline repo to 10.3.2.1 and DocDB to 5.0.0

addtionally, use the newer rds-ca-rsa2048-g1 CA certs
as the original rds-ca-2019 CA certs have expired.

rondeau-aws commented on integ/components/deadline/deadline_01_repository/scripts/bastion/testing/DL-read-docdb-response.sh:
```diff
@@ -38,7 +38,8 @@ else
-mongo --quiet --ssl --host="$DB_ADDRESS" --sslCAFile="$CERT_CA" --username="$DB_USERNAME" --password="$DB_PASS" --eval='printjson( db.adminCommand( { listDatabases: 1, nameOnly: true, filter: { "name": "deadline10db" } } ) )'
-
+# Mongo command to query for "deadline10db" database.
+# We remove any Timestamp fields returned from the EJSON returned by `mongo`
+# with the `delete` statement.
```
```suggestion
# We delete Timestamp fields from the EJSON result to avoid errors when parsing
# them in Python.
```
